Gem/Code/Source/Components/NetworkTeleportCompatibleComponent.cpp

```diff
@@ -0,0 +1,73 @@
+#include <Source/Components/NetworkTeleportCompatibleComponent.h>
+
+#include <Multiplayer/Components/NetworkTransformComponent.h>
+#include <AzCore/Component/TransformBus.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzFramework/Physics/RigidBodyBus.h>
+
+namespace MultiplayerSample
+{
+    void NetworkTeleportCompatibleComponent::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<NetworkTeleportCompatibleComponent, NetworkTeleportCompatibleComponentBase>()
+                ->Version(1);
+        }
+        NetworkTeleportCompatibleComponentBase::Reflect(context);
+    }
+
+    void NetworkTeleportCompatibleComponent::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void NetworkTeleportCompatibleComponent::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    // Controller
+
+    NetworkTeleportCompatibleComponentController::NetworkTeleportCompatibleComponentController(NetworkTeleportCompatibleComponent& parent)
+        : NetworkTeleportCompatibleComponentControllerBase(parent)
+    {
+    }
+
+    void NetworkTeleportCompatibleComponentController::OnActivate(
+        [[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void NetworkTeleportCompatibleComponentController::OnDeactivate(
+        [[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
```

Comment:
Seems like these don't need to be overridden.

Reply:
d'oh, forget to consolidate these in the header (seems we at least need them there since they're virtual)

Reply:
Ah yeah, some of those are pure. Handling them in the header sounds good.